tree/ntuple/v7/inc/ROOT/RMiniFile.hxx

```diff
@@ -0,0 +1,221 @@
+
+The class points to the header and footer keys, which in turn have the references to the pages.
+Only the RNTuple key will be listed in the list of keys. Like TBaskets, the pages are "invisible" keys.
+Byte offset references in the RNTuple header and footer reference directly the data part of page records,
+skipping the TFile key part.
+*/
+// clang-format on
+struct RNTuple {
+   /// Allows for evolving the struct in future versions
+   std::uint32_t fVersion = 0;
+   /// Allows for skipping the struct
+   std::uint32_t fSize = sizeof(RNTuple);
+   /// The file offset of the header excluding the TKey part
+   std::uint64_t fSeekHeader = 0;
+   /// The size of the compressed ntuple header
+   std::uint32_t fNBytesHeader = 0;
+   /// The size of the uncompressed ntuple header
+   std::uint32_t fLenHeader = 0;
+   /// The file offset of the footer excluding the TKey part
+   std::uint64_t fSeekFooter = 0;
+   /// The size of the compressed ntuple footer
+   std::uint32_t fNBytesFooter = 0;
+   /// The size of the uncompressed ntuple footer
+   std::uint32_t fLenFooter = 0;
+   /// Currently unused, reserved for later use
+   std::uint64_t fReserved = 0;
+
+   bool operator ==(const RNTuple &other) const {
+      return fVersion == other.fVersion &&
+         fSize == other.fSize &&
+         fSeekHeader == other.fSeekHeader &&
+         fNBytesHeader == other.fNBytesHeader &&
+         fLenHeader == other.fLenHeader &&
+         fSeekFooter == other.fSeekFooter &&
+         fNBytesFooter == other.fNBytesFooter &&
+         fLenFooter == other.fLenFooter &&
+         fReserved == other.fReserved;
+   }
+};
+
+namespace Internal {
+
+/// Holds status information of an open ROOT file during writing
+struct RTFileControlBlock;
+
+// clang-format off
+/**
+\class ROOT::Experimental::Internal::RMiniFileReader
+\ingroup NTuple
+\brief Read RNTuple data blocks from a TFile container, provided by a RRawFile
+
+A RRawFile is used for the byte access.  The class implements a minimal subset of TFile, enough to extract
+RNTuple data keys.
+*/
+// clang-format on
+class RMiniFileReader {
+private:
+   /// The raw file used to read byte ranges
+   Detail::RRawFile *fFile = nullptr;
+   /// Indicates whether the file is a TFile container or an RNTuple raw file
+   bool fIsRaw = false;
+   /// Used when the file turns out to be a raw file
+   RNTuple GetNTupleRaw(std::string_view ntupleName);
+   /// Used when the file turns out to be a TFile container
+   RNTuple GetNTupleProper(std::string_view ntupleName);
+
+public:
+   RMiniFileReader() = default;
+   /// Uses the given raw file to read byte ranges
+   explicit RMiniFileReader(Detail::RRawFile *rawFile);
+   /// Extracts header and footer location for the RNTuple identified by ntupleName
+   RNTuple GetNTuple(std::string_view ntupleName);
+   /// Reads a given byte range from the file into the provided memory buffer
+   void ReadBlob(void *buffer, size_t nbytes, std::uint64_t offset);
```

Comment:
Isn't that similar to other ReadBuffer or is it different enough to justify a different name?

Reply:
Renamed to `ReadBuffer`